### PR TITLE
Integrate @RunOnVirtualThread with the Quarkus scheduler

### PR DESCRIPTION
**Pull request by cescoffier**
Allows @Scheduled method to run on a virtual thread.

### CODE REVIEW

**Review by mkouba — COMMENTED** [review]
Looks good but I think that we should also:

1. Update the `io.quarkus.quartz.runtime.QuartzSchedulerImpl.InvokerJob.execute()`,
2. Enhance the programmatic scheduling API; maybe add a `Scheduler.JobDefinition.setTask(Consumer<ScheduledExecution>)` variant with a boolean param (`runOnVirtualThread` or something like that).

**Comment by cescoffier** [comment]
I need to look at Quartz, I vaguely remember that they have their own threads.

**Comment by mkouba** [comment]
> I need to look at Quartz, I vaguely remember that they have their own threads.

We don't use them anymore to execute the logic (merely to execute the trigger) ... https://github.com/quarkusio/quarkus/blob/main/extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzSchedulerImpl.java#L859-L904 

(and you participated in this change: https://github.com/quarkusio/quarkus/commit/9452ab3000869fe2b72b782ead056dbf71fe6502 ;-)

**Comment by cescoffier** [comment]
@mkouba I've:

- extended the doc a bit
- done the work for Quartz and added ITs
- followed your idea for the programmatic API.

File: extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java [hunk 1]
```diff
@@ -386,20 +387,36 @@ void execute(ZonedDateTime now, Vertx vertx) {
                 return;
             }
             ZonedDateTime scheduledFireTime = trigger.evaluate(now);
+            Context context = VertxContext.getOrCreateDuplicatedContext(vertx);
+            VertxContextSafetyToggle.setContextSafe(context, true);
```
Comment (mkouba) [inline extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java:391]: Why have you moved this code outside the `if (scheduledFireTime != null)` branch? If `scheduledFireTime == null` then the job should not be executed at all and so it does not make sense to call `VertxContextSafetyToggle.setContextSafe()`...

**Review by mkouba — APPROVED** [review]
Looks really good, thanks!

I've added one minor comment...

**Review by mkouba — APPROVED** [review]

**Pull request merged**